src/pages/proposal.tsx

```diff
@@ -252,138 +338,131 @@ export default function ProposalPage() {
                   onUpdateDeleted={refetchUpdates}
                 />
               )}
+              {proposal && (
+                <>
+                  <SurveyResults
+                    votes={votes}
+                    isLoadingVotes={votesState.loading}
+                    surveyTopics={surveyTopics}
+                    isLoadingSurveyTopics={isLoadingSurveyTopics}
+                  />
+                  <ProposalResults
+                    proposal={proposal}
+                    votes={votes}
+                    partialResults={partialResults}
+                    loading={proposalState.loading || votesState.loading}
+                    onOpenVotesList={() => updatePageState({ showVotesList: true })}
+                    elementRef={proposalResults}
+                  />
+                </>
+              )}
               <ProposalComments proposal={proposal} loading={proposalState.loading} />
             </Grid.Column>
 
             <Grid.Column tablet="4" className="ProposalDetailActions">
               {!!proposal?.vesting_address && <VestingContract vestingAddress={proposal.vesting_address} />}
               {proposal && <ProposalCoAuthorStatus proposalId={proposal.id} proposalFinishDate={proposal.finish_at} />}
               <div className="ProposalDetail__StickySidebar">
-                <ForumButton
-                  loading={proposalState.loading}
-                  disabled={!proposal}
-                  href={(proposal && forumUrl(proposal)) || ''}
-                />
-                <SubscribeButton
-                  loading={proposalState.loading || subscriptionsState.loading || subscribing}
-                  disabled={!proposal}
-                  subscribed={subscribed}
-                  onClick={() => subscribe(!subscribed)}
-                />
                 {showProposalUpdatesActions && (
                   <ProposalUpdatesActions
                     nextUpdate={nextUpdate}
                     currentUpdate={currentUpdate}
                     onPostUpdateClick={handlePostUpdateClick}
                   />
                 )}
-                <ProposalResultSection
+                <ProposalGovernanceSection
                   disabled={!proposal || !votes}
-                  loading={voting || proposalState.loading || votesState.loading}
+                  loading={proposalState.loading || votesState.loading}
                   proposal={proposal}
                   votes={votes}
-                  changingVote={options.changing}
-                  onChangeVote={(_, changing) => patchOptions({ changing })}
-                  onOpenVotesList={() => patchOptions({ showVotesList: true })}
-                  onVote={(_, choice, choiceIndex) => vote(choice, choiceIndex)}
+                  partialResults={partialResults}
+                  choices={choices}
+                  castingVote={castingVote}
+                  onChangeVote={(_, changing) => updatePageState({ changingVote: changing })}
```

Comment:
If we are using the handler pattern in all other function props here I'd move this to a `handleChangeVote` function.

Reply:
We are not jiji. I think that most functions that are just an update on the state are all anonymous functions, and if there are more steps then they are extracted to a handler.

Some examples:


```typescript
onOpenVotesList={() => updatePageState({ showVotesList: true })}
onClose={() => updatePageState({ showVotesList: false })}
onClose={() => updatePageState({ confirmSubscription: false })}
onClose={() => updatePageState({ confirmDeletion: false })}
```

I could extract them all to handlers though. WDYT? 
I don't mind them inline also, because it's getting long to read and declarations will be far up from usages.

Reply:
I don't mind. It's just weird to read handlers and inline functions in the same component props.